cloudbridge/cloud/providers/gce/resources.py

```diff
@@ -1818,16 +1831,46 @@ def id(self):
     def name(self):
         """
         Get the snapshot name.
-         """
+        """
         return self._snapshot.get('name')
 
+    @name.setter
+    # pylint:disable=arguments-differ
+    def name(self, value):
+        GCESnapshot.assert_valid_resource_name(value)
+        # In GCE, the name of the snapshot is provided by the client when
+        # initially creating the resource. The name cannot be changed after
+        # the snapshot is created.
+        cb.log.warning("Setting snapshot name after it is created is not "
+                       "supported by this provider.")
+
```

Comment:
Same as for Volume name. The name should ideally be editable. In GCE, I guess the 'name' maps more to cloudbridge's id.